### PR TITLE
Feature/spark df transformation

### PR DESCRIPTION
**Pull request by ahmadnazeri**
# Description
Adding support for Spark DF transformations 

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [x] I have fixed any merge conflicts

### CODE REVIEW

File: provider/scripts/offline_store_spark_runner.py [hunk 1]
```diff
@@ -71,6 +75,27 @@ def execute_df_job(output_uri, code, sources):
         print(f"Issue with execution of the transformation: {e}")
         raise e
 
+def get_code_from_file(file_path, aws_region=None):
+    prefix_len = len("s3://")
+    if "s3://" == file_path[:prefix_len]:
+        split_path = file_path[prefix_len:].split("/")
```
Comment (sdreyer) [inline provider/scripts/offline_store_spark_runner.py:81]: Can you break this function up more or add comments. Unclear whats going on by looking at it

File: provider/spark.go [hunk 2]
```diff
@@ -811,6 +826,30 @@ func (spark *SparkOfflineStore) sqlTransformation(config TransformationConfig, i
...
+	} else if isUpdate && !exists {
+		return fmt.Errorf("transformation %v doesn't exist at %s and you are trying to update", config.TargetTableID, transformationDestination)
+	}
+
+	sparkArgs, err := spark.getDFArgs(transformationDestination, config.Query, spark.Store.Region(), config.SourceMapping)
+	if err != nil {
+		return fmt.Errorf("error with getting df arguments %v", sparkArgs)
+	}
+
+	if err := spark.Executor.RunSparkJob(sparkArgs); err != nil {
```
Comment (sdreyer) [inline provider/spark.go:847]: How does this work atm? Is it creating files with new timestamps for each update?

File: provider/scripts/tests/conftest.py [hunk 3]
```diff
@@ -34,8 +34,8 @@ def sql_invalid_local_arguments():
 
 @pytest.fixture(scope="module")
```
Comment (sdreyer) [inline provider/scripts/tests/conftest.py:35]: Add this test to the pytest target in the makefil if you havent already

File: provider/spark.go (hunk 2 above)
Comment (ahmadnazeri) [reply provider/spark.go:847]: yes, it creates a new timestamp with each update. The design is that we only worry about the latest timestamp.

File: provider/scripts/offline_store_spark_runner.py [hunk 4]
```diff
@@ -71,6 +75,40 @@ def execute_df_job(output_uri, code, sources):
...
+        file_path: string (path to file)
+        aws_region: string (aws region where s3 bucket is located)
+    Return:
+        code: code object that could be executed
+    """
+    
+    prefix_len = len("s3://")
+    code = None
+    if "s3://" == file_path[:prefix_len]:
+        split_path = file_path[prefix_len:].split("/")
```
Comment (sdreyer) [inline provider/scripts/offline_store_spark_runner.py:93]: Can you add a comment here to explain what these path splitting functions are doing?

**Review by sdreyer — APPROVED** [review]

**Pull request merged**